### PR TITLE
Create navigator widget factory

### PR DESCRIPTION
**Pull request by DoroNahari**
Signed-off-by: Doron Nahari <doron.nahari@sap.com>

#### What it does

This pr converts the navigator widget creation to be with a new navigator widget factory which is extensible and so we can extend it and configure it as desired.

#### How to test

extend:
```typescript
@injectable()
export class CustomNavigatorWidget extends NavigatorWidgetFactory {
    async createWidget(): Promise<ViewContainer> {
        this.navigatorWidgetOptions = {
            canHide: true,
            initiallyCollapsed: true,
        };
        return super.createWidget();
    }
}
```
#### Review checklist

- [X] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/navigator/src/browser/navigator-frontend-module.ts [hunk 1]
```diff
@@ -35,8 +35,9 @@ import { NavigatorDiff } from './navigator-diff';
 import { NavigatorLayoutVersion3Migration } from './navigator-layout-migrations';
 import { NavigatorTabBarDecorator } from './navigator-tab-bar-decorator';
 import { TabBarDecorator } from '@theia/core/lib/browser/shell/tab-bar-decorator';
+import { NavigatorWidgetFactory } from './navigator-widget-factory';
 
-export default new ContainerModule(bind => {
+export default new ContainerModule((bind, unbind, isBound, rebind) => {
```
Comment (amiramw) [inline packages/navigator/src/browser/navigator-frontend-module.ts:40]: why do you need unbind, isBound and rebind?

**Review by amiramw — COMMENTED** [review]
BTW, how to test can also be: 

extend:

```typescript
@injectable()
export class CustomNavigatorWidget extends NavigatorWidgetFactory {
    protected fileNavigatorWidgetOptions: ViewContainer.Factory.WidgetOptions = {
        canHide: true,
        initiallyCollapsed: true,
    }
}
```

File: packages/navigator/src/browser/navigator-frontend-module.ts (hunk 1 above)
Comment (DoroNahari) [reply packages/navigator/src/browser/navigator-frontend-module.ts]: Leftovers from my testing. removed

**Review by amiramw — APPROVED** [review]

**Comment by vince-fugnitto** [comment]
@amiramw would you like to merge the pull-request for tomorrow's release?

**Comment by amiramw** [comment]
@vince-fugnitto yes, we prefer.

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto yes, we prefer.

I'll let you merge, please go ahead :)

**Pull request merged**